.github/workflows/register-schemas-dryrun.yaml

```diff
@@ -0,0 +1,53 @@
+on: [pull_request]
+
+name: Register schemas (dry run)
+
+env:
+  RENV_PATHS_ROOT: ~/.local/share/renv
+
+jobs:
+  register-schemas-dryrun:
+    runs-on: macos-10.15
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: r-lib/actions/setup-r@master
+        with:
+          r-version: '4.0'
+
+      - name: Cache packages
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.RENV_PATHS_ROOT }}
+          key: ${{ runner.os }}-renv-${{ hashFiles('**/renv.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-renv-
+
+      - name: Restore packages
+        shell: Rscript {0}
+        run: |
+          if (!requireNamespace("renv", quietly = TRUE)) install.packages("renv")
+          renv::restore()
```

Comment:
Why are the cache and restore steps necessary?

Reply:
The caching is just to reduce the time the job takes to run by not re-installing every R package every time. `renv::restore()` looks at the package lockfile and makes sure that the correct versions of the packages are available, and installs them if not -- so it's mainly for reproducibility of the script.